test/common_nn.py

```diff
@@ -364,12 +364,29 @@ def multilabelmarginloss_reference(input, target, size_average=True, reduce=True
     return output / dim
 
 
+def hingeembeddingloss_reference(input, target, margin=1.0, size_average=True, reduce=True):
+    # needed for legacy tests
+    if not isinstance(input, Variable):
```

Comment:
Why is this needed for the legacy tests?

Reply:
torch.where function only works for Variables. I also tried using the Tensor.map_, which worked for the legacy test but broke test_nn. Let me know if you think there's a better way to do this though.